blog: v4 (#1078)
simplex-chat/simplex-chat: blog/20220928-simplex-chat-v4-encrypted-database.md

The reason for that problem was that some network operations required for group connections were not retried. This release improves stability of all network operations - receiving messages, making new contact connections, receiving files and connecting to members in the groups you join.

### Deleting files and media

<img src="./images/20220928-files-media.png" width="330">

While the database with messages and all contacts is now encrypted, the files you receive and send are not (we are planning to improve it later). To protect the security of the files we have added an option to delete all files from the app storage - just make sure not to do it before the files you send are delivered to the recipients, or they won't receive them.

### For developers - TypeScript SDK for integrating with SimpleX Chat

It's been quite some time since it's been possible to create a chat-bot using SimpleX Chat as a library - but you would have to write Haskell code, and this is not the most widely used language.

> With v4 we are announcing [TypeScript SimpleX Chat Client SDK](https://github.com/simplex-chat/simplex-chat/tree/stable/packages/simplex-chat-client/typescript) that you can use to create for any integrations with SimpleX Chat CLI, that can be used as a local WebSockets server.

Reviewer: ```suggestion
With v4 we are announcing [TypeScript SimpleX Chat Client SDK](https://github.com/simplex-chat/simplex-chat/tree/stable/packages/simplex-chat-client/typescript) that you can use for integrations with SimpleX Chat CLI.
```

> You can run SimpleX Chat CLI as a local WebSockets server on any port, we use 5225 here:

Reviewer: ```suggestion
You can run SimpleX Chat CLI as a local WebSockets server on any port, e.g.:
```

```bash
simplex-chat -p 5225
```

> Then you can make a simple JavaScript or TypeScript application that would connect to it and control it via simple WebSocket API that has all necessary types and convenience functions defined in TypeScript SDK. See this [sample bot](https://github.com/simplex-chat/simplex-chat/blob/stable/packages/simplex-chat-client/typescript/examples/squaring-bot.js) you can start from.

Reviewer: simple x2
```suggestion
Then you can make a simple JavaScript or TypeScript application that would connect to it and control it via WebSocket API that has all necessary types and convenience functions defined in TypeScript SDK. See this [sample bot](https://github.com/simplex-chat/simplex-chat/blob/stable/packages/simplex-chat-client/typescript/examples/squaring-bot.js) you can start from.
```

---
layout: layouts/article.html
title: "SimpleX Chat v4.0 with encrypted database is released"
date: 2022-09-28
permalink: "/blog/20220928-simplex-chat-v4-encrypted-database.html"
---

# SimpleX Chat v4.0 with encrypted database is released

**Published:** Sep 28, 2022

## What's new in v4.0

> - [encrypted chat database](#chat-database-encrypted-with-passphrase).
> - [improved stability of creating new connections: more reliable groups, files and contacts](#improved-stability-of-creating-new-connections).
> - [deleting files and media](#deleting-files-and-media)
> - [support for self-hosted WebRTC STUN/TURN servers](#self-hosted-webrtc-stunturn-servers)
> - [TypeScript SDK for integrating with SimpleX Chat](#typescript-sdk-for-integrating-with-simplex-chat) (e.g., chat bots or chat assistants).
> - support animated images in Android app.
> - disable notifications per contact / group in terminal app (it is already supported in mobile apps).

Reviewer: ```suggestion
- [encrypted chat database](#chat-database-encrypted-with-passphrase).
- [improved stability of creating new connections: more reliable groups, files and contacts](#improved-stability-of-creating-new-connections).
- [deleting files and media](#deleting-files-and-media).
- [support for self-hosted WebRTC STUN/TURN servers](#self-hosted-webrtc-stunturn-servers).
- [TypeScript SDK for integrating with SimpleX Chat](#typescript-sdk-for-integrating-with-simplex-chat) (e.g., chat bots or chat assistants).
- support animated images in Android app.
- disable notifications per contact / group in terminal app (it is already supported in mobile apps).
```

Other new features since v3.0:

> - secret chat groups (see details in [v3.1 announcement](./20220808-simplex-chat-v3.1-chat-groups.md#secret-chat-groups) - they are fully decentralized, only their members know these groups exist.

Reviewer: ```suggestion
- secret chat groups (see details in [v3.1 announcement](./20220808-simplex-chat-v3.1-chat-groups.md#secret-chat-groups) - fully decentralized, only their members know these groups exist.
```
